### PR TITLE
Feature/offline form support

### PR DESCRIPTION
**Pull request by sagarnasit**
Fixes: https://github.com/rtCamp/labs/issues/10

### CODE REVIEW

File: inc/classes/class-offline-form.php [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	}
+
+	/**
+	 * Setup actions/filters
+	 *
+	 * @return void
+	 */
+	protected function setup_hooks() {
+
+		add_action( 'wp_front_service_worker', array( $this, 'offline_form_service_worker' ), 11 );
```
Comment (deepaklalwani97) [inline inc/classes/class-offline-form.php]: Is there any specific reason to use 11 priority here. If there is we should specify in a comment

File: inc/classes/class-admin-settings.php [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+		add_action( 'admin_menu', array( $this, 'register_options_page' ) );
+	}
+
+	/**
+	 * Register plugin setting.
+	 */
+	public function register_plugin_settings() {
+		$args = array(
+			'type'              => 'string',
+			'description'       => __( 'Form routes for offline form submission.', 'pwa-extension' ),
```
Comment (deepaklalwani97) [inline inc/classes/class-admin-settings.php]: Use `esc_html__` instead of `__`

File: inc/classes/class-admin-settings.php [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+			'sanitize_callback' => array( $this, 'validate_setting' ),
+		);
+
+		register_setting( $this->page_slug, 'rt_pwa_extension_options', $args );
+		add_settings_section( 'pwa-extension-setting-options', __( 'Form Routes Options', 'pwa-extension' ), array( $this, 'callback_pwa_setting_section' ), 'options-general.php' );
```
Comment (deepaklalwani97) [inline inc/classes/class-admin-settings.php]: Use `esc_html__` instead of `__`

File: inc/classes/class-admin-settings.php [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+		add_settings_field( 'pwa-extension-routes-input', __( 'Form Routes', 'pwa-extension' ), array( $this, 'callback_url_input' ), 'options-general.php', 'pwa-extension-setting-options' );
```
Comment (deepaklalwani97) [inline inc/classes/class-admin-settings.php]: Use `esc_html__` instead of `__`

File: inc/classes/class-admin-settings.php [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+		$value = get_option( 'rt_pwa_extension_options' );
+		printf( '<textarea name="rt_pwa_extension_options" rows="5">%1$s</textarea>', esc_textarea( $value ) );
+		printf( '<p><i>%1$s</i></p>', esc_html__( 'Add multiple routes in separate new line.', 'pwa-extension' ) );
+	}
+
+	/**
+	 * Register plugin option page.
+	 */
+	public function register_options_page() {
+		add_options_page( __( 'PWA Extension Settings', 'pwa-extension' ), __( 'PWA Extension Settings', 'pwa-extension' ), 'manage_options', $this->page_slug, array( $this, 'options_page' ) );
```
Comment (deepaklalwani97) [inline inc/classes/class-admin-settings.php]: Use `esc_html__` instead of `__`

File: inc/classes/class-admin-settings.php [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+	 * @return array.
+	 */
+	public function validate_setting( $input ) {
+
+		$validated = sanitize_textarea_field( $input );
+
+		// Set error message when invalid.
+		if ( $validated !== $input ) {
+			$type    = 'error';
+			$message = __( 'Invalid \'Form Route\'. Please enter valid form routes.', 'pwa-extension' );
```
Comment (deepaklalwani97) [inline inc/classes/class-admin-settings.php]: Use `esc_html__` instead of `__`

File: inc/classes/class-admin-settings.php [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+					break;
+				}
+			}
+
+			// Set error message for white spaces.
+			if ( true === $white_space ) {
+				add_settings_error(
+					'rt_pwa_extension_options',
+					'pwa-extension-white-space-not-allowed',
+					__( 'White space are not allowed in \'Form Routes\'.', 'pwa-extension' ),
```
Comment (deepaklalwani97) [inline inc/classes/class-admin-settings.php]: Use `esc_html__` instead of `__`

**Review by deepaklalwani97 — CHANGES_REQUESTED** [review]

**Comment by sagarnasit** [comment]
@deepaklalwani97 Added a few changes. Also haven't use `esc_html__` instead of `__` as all the places are static strings and do not need explicit escaping of Html.

**Comment by deepaklalwani97** [comment]
@sagarnasit When those static strings are translated the HTML content can be injected with translation text that's why we use `esc_html__` with static strings also.

**Review by deepaklalwani97 — APPROVED** [review]

**Comment by sagarnasit** [comment]
@deepaklalwani97 Added escaping changes.

**Pull request merged**